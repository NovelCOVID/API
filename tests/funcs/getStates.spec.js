const axios = require('axios')
const redis = require('ioredis')

const fixture = require('../worldometer-states-fixture')
const getStates = require('../../funcs/getStates')

jest.mock('ioredis')
jest.mock('axios')
redis.set = jest.fn()

describe('getStatesAPI API', () => {
  beforeEach(() => {
    axios.get.mockClear()
    redis.set.mockClear()
  })

  it('scrapes for states data', async () => {
    axios.get.mockResolvedValue({ data: fixture, status: 200 })

    const result = await getStates([], redis)
    // Filter out 'updated', since this will always change between runs.
    const { updated, ...rest } = result

    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(redis.set).toHaveBeenCalledTimes(1)
    expect(result).toHaveLength(53)
    expect(rest).toMatchSnapshot()
  })
})
