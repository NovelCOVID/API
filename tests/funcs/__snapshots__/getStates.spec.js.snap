// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStatesAPI API scrapes for states data 1`] = `
Object {
  "0": Object {
    "cases": 25665,
    "deaths": 183,
    "state": "New York",
    "todayCases": 4790,
    "todayDeaths": 26,
  },
  "1": Object {
    "cases": 2844,
    "deaths": 27,
    "state": "New Jersey",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "10": Object {
    "cases": 777,
    "deaths": 9,
    "state": "Massachusetts",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "11": Object {
    "cases": 720,
    "deaths": 7,
    "state": "Colorado",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "12": Object {
    "cases": 644,
    "deaths": 6,
    "state": "Pennsylvania",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "13": Object {
    "cases": 615,
    "deaths": 2,
    "state": "Tennessee",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "14": Object {
    "cases": 442,
    "deaths": 6,
    "state": "Ohio",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "15": Object {
    "cases": 416,
    "deaths": 5,
    "state": "Wisconsin",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "16": Object {
    "cases": 415,
    "deaths": 10,
    "state": "Connecticut",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "17": Object {
    "cases": 410,
    "deaths": 0,
    "state": "North Carolina",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "18": Object {
    "cases": 365,
    "deaths": 12,
    "state": "Indiana",
    "todayCases": 106,
    "todayDeaths": 5,
  },
  "19": Object {
    "cases": 299,
    "deaths": 5,
    "state": "South Carolina",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "2": Object {
    "cases": 2221,
    "deaths": 110,
    "state": "Washington",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "20": Object {
    "cases": 288,
    "deaths": 3,
    "state": "Maryland",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "21": Object {
    "cases": 257,
    "deaths": 1,
    "state": "Utah",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "22": Object {
    "cases": 254,
    "deaths": 7,
    "state": "Virginia",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "23": Object {
    "cases": 249,
    "deaths": 1,
    "state": "Mississippi",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "24": Object {
    "cases": 245,
    "deaths": 4,
    "state": "Nevada",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "25": Object {
    "cases": 235,
    "deaths": 1,
    "state": "Minnesota",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "26": Object {
    "cases": 234,
    "deaths": 2,
    "state": "Arizona",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "27": Object {
    "cases": 197,
    "deaths": 0,
    "state": "Arkansas",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "28": Object {
    "cases": 196,
    "deaths": 5,
    "state": "Missouri",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "29": Object {
    "cases": 196,
    "deaths": 0,
    "state": "Alabama",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "3": Object {
    "cases": 2220,
    "deaths": 43,
    "state": "California",
    "todayCases": 87,
    "todayDeaths": 3,
  },
  "30": Object {
    "cases": 191,
    "deaths": 5,
    "state": "Oregon",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "31": Object {
    "cases": 137,
    "deaths": 2,
    "state": "District of Columbia",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "32": Object {
    "cases": 124,
    "deaths": 4,
    "state": "Kentucky",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "33": Object {
    "cases": 107,
    "deaths": 0,
    "state": "Maine",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "34": Object {
    "cases": 106,
    "deaths": 0,
    "state": "Rhode Island",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "35": Object {
    "cases": 105,
    "deaths": 0,
    "state": "Iowa",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "36": Object {
    "cases": 101,
    "deaths": 1,
    "state": "New Hampshire",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "37": Object {
    "cases": 87,
    "deaths": 0,
    "state": "Delaware",
    "todayCases": 19,
    "todayDeaths": 0,
  },
  "38": Object {
    "cases": 83,
    "deaths": 0,
    "state": "New Mexico",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "39": Object {
    "cases": 82,
    "deaths": 2,
    "state": "Kansas",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "4": Object {
    "cases": 1328,
    "deaths": 15,
    "state": "Michigan",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "40": Object {
    "cases": 81,
    "deaths": 2,
    "state": "Oklahoma",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "41": Object {
    "cases": 77,
    "deaths": 1,
    "state": "Hawaii",
    "todayCases": 0,
    "todayDeaths": 1,
  },
  "42": Object {
    "cases": 75,
    "deaths": 5,
    "state": "Vermont",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "43": Object {
    "cases": 61,
    "deaths": 0,
    "state": "Nebraska",
    "todayCases": 11,
    "todayDeaths": 0,
  },
  "44": Object {
    "cases": 50,
    "deaths": 0,
    "state": "Idaho",
    "todayCases": 3,
    "todayDeaths": 0,
  },
  "45": Object {
    "cases": 45,
    "deaths": 0,
    "state": "Montana",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "46": Object {
    "cases": 36,
    "deaths": 0,
    "state": "Alaska",
    "todayCases": 4,
    "todayDeaths": 0,
  },
  "47": Object {
    "cases": 32,
    "deaths": 0,
    "state": "North Dakota",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "48": Object {
    "cases": 28,
    "deaths": 1,
    "state": "South Dakota",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "49": Object {
    "cases": 26,
    "deaths": 0,
    "state": "Wyoming",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "5": Object {
    "cases": 1285,
    "deaths": 12,
    "state": "Illinois",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "50": Object {
    "cases": 20,
    "deaths": 0,
    "state": "West Virginia",
    "todayCases": 4,
    "todayDeaths": 0,
  },
  "51": Object {
    "cases": 49,
    "deaths": 0,
    "state": "Diamond Princess Cruise",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "52": Object {
    "cases": 30,
    "deaths": 1,
    "state": "Grand Princess Cruise",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "6": Object {
    "cases": 1227,
    "deaths": 18,
    "state": "Florida",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "7": Object {
    "cases": 1172,
    "deaths": 35,
    "state": "Louisiana",
    "todayCases": 0,
    "todayDeaths": 0,
  },
  "8": Object {
    "cases": 826,
    "deaths": 9,
    "state": "Texas",
    "todayCases": 20,
    "todayDeaths": 0,
  },
  "9": Object {
    "cases": 803,
    "deaths": 26,
    "state": "Georgia",
    "todayCases": 0,
    "todayDeaths": 0,
  },
}
`;
